python/pyspark/ml/regression.py

```diff
@@ -1116,7 +1116,7 @@ def setParams(self, featuresCol="features", labelCol="label", predictionCol="pre
                   maxDepth=5, maxBins=32, minInstancesPerNode=1, minInfoGain=0.0,
                   maxMemoryInMB=256, cacheNodeIds=False, subsamplingRate=1.0,
                   checkpointInterval=10, lossType="squared", maxIter=20, stepSize=0.1, seed=None,
-                  impuriy="variance", featureSubsetStrategy="all"):
+                  impurity="variance", featureSubsetStrategy="all"):
```

Comment:
I think this can't just changed like this since it's going to break other users codes.

Reply:
is anyone depending on the typoed version?

Reply:
Of course that's possible - user upgraded Spark and suddenly it gives no such keyword exception.

Reply:
what about this until next major release?
```
    def setParams(self, featuresCol="features", labelCol="label", predictionCol="prediction",
                  maxDepth=5, maxBins=32, minInstancesPerNode=1, minInfoGain=0.0,
                  maxMemoryInMB=256, cacheNodeIds=False, subsamplingRate=1.0,
                  checkpointInterval=10, lossType="squared", maxIter=20, stepSize=0.1, seed=None,
                  impuriy=None, impurity="variance", featureSubsetStrategy="all"):
        if impuriy is not None: # for backward compatibility
        	impurity = impuriy
```

Reply:
@HyukjinKwon I think impuriy need modified.
if someone set`impurity` in `GBTRegressor` function, will get error. because `impuriy` is one of the params in `setParams` function.
`gbt = GBTRegressor(..., impurity="variance")`

if someone to set impuriy as impurity, also will get error. because `__init__ ` function have no `impuriy` param, it only hava  impurity.
`gbt = GBTRegressor(..., impuriy="variance")`

Maybe no one used pyspark to set the parameter of GBTRegressor impurity.

Reply:
Can you open a PR with explaining what happens before after the change as you described?